File: app/src/main/java/io/ipoli/android/reminder/ReminderMinutesParser.java
package io.ipoli.android.reminder;

import android.support.v4.util.Pair;

/**
 * Created by Polina Zhelyazkova <polina@ipoli.io>
 * on 6/30/16.
 */
public class ReminderMinutesParser {

    public static Pair<Long, TimeOffsetType> parseCustomMinutes(long minutes) {
        if(minutes == 0) {
            return new Pair<>(minutes, TimeOffsetType.MINUTES);
        }

        TimeOffsetType[] types = TimeOffsetType.values();
        for (int i = types.length - 1; i >= 0; i--) {
            TimeOffsetType type = types[i];
            if (minutes % type.getMinutes() == 0) {
                return new Pair<>(minutes / type.getMinutes(), type);
            }
        }

        return null;
    }
}

File: app/src/main/java/io/ipoli/android/common/parser/ReminderMinutesParser.kt
package io.ipoli.android.common.parser

import io.ipoli.android.quest.reminder.picker.TimeUnit

/**
 * Created by Venelin Valkov <venelin@mypoli.fun>
 * on 10/4/17.
 */
object ReminderMinutesParser {

    fun parseCustomMinutes(minutes: Long): Pair<Long, TimeUnit> {
        val timeUnit = TimeUnit.values().reversed()
            .firstOrNull { minutes % it.minutes == 0L }

        return if (timeUnit == null) {
            Pair(minutes, TimeUnit.MINUTES)
        } else {
            Pair(minutes / timeUnit.minutes, timeUnit)
        }
    }
}